File: arrays/convert-1d-array-into-2d-array.cs
public class Solution {
    public int[][] Construct2DArray(int[] original, int m, int n) {
        if (original.Length != m * n) return [];

        int[][] result = new int[m][];
        for (int i = 0; i < result.Length; ++i) {
            result[i] = new int[n];
        }

        for (int i = 0; i < original.Length; ++i) {
            result[i / n][i % n] = original[i];
        }

        return result;
    }
}

File: arrays/convert-1d-array-into-2d-array.rs
impl Solution {
    pub fn construct2_d_array(original: Vec<i32>, m: i32, n: i32) -> Vec<Vec<i32>> {
        let m = m as usize;
        let n = n as usize;
        if original.len() != m * n {
            return Vec::new();
        }

        let mut result = vec![vec![0; n]; m];
        for i in 0..original.len() {
            result[i / n][i % n] = original[i];
        }

        result
    }
}
